src/libsync/propagateuploadng.cpp

```diff
@@ -219,7 +224,9 @@ void PropagateUploadFileNG::slotDeleteJobFinished()
 
 void PropagateUploadFileNG::startNewUpload()
 {
-    ASSERT(propagator()->_activeJobList.count(this) == 1);
+    if (propagator()->_activeJobList.count(this) == 0)
```

Comment:
So you are changing the invariant?  I guess this is only hapenning when there were more data on the server.  But i'd argue that we should be consistant, and always add the activeJobList in the same place.

For example, if we start uploading chunks in parallel again, code like that can lead to inconsistency.